lib/mail_room/mailbox.rb

```diff
@@ -66,7 +67,14 @@ def initialize(attributes={})
     end
 
     def logger
-      @logger ||= MailRoom::Logger::Structured.new(structured_logger_file_name)
+      @logger ||=
+        case self[:logger]
+          when Logger
+            self[:logger]
+          else
+            self[:logger] ||= {}
+            MailRoom::Logger::Structured.new(self[:logger][:log_path])
```

Comment:
If the `:log_path` is not set, the docs suggest we'd default to STDOUT, but in this case the logger is setup with `nil`. If I'm not mistaken this doesn't work.

```ruby
require 'logger'

Logger.new(nil) << "foo"
# => nil
Logger.new($stdout) << "foo"
foo # => 3 # number of bytes written
```

I think this should be:

```ruby
sink = self[:logger][:log_path] || $stdout
MailRoom::Logger::Structured.new(sink)
```

Reply:
@tpitale I might be wrong with that note, as it's a time ago I created the original PR that added logging. Let me know what you think about this comment though! :smile:

Reply:
I’ll have to update the readme. We don’t want the default to be stdout, I don’t want the default behavior for existing users to change, which means this is entirely opt-in.

Reply:
I couldn't find where in the docs it says otherwise, so I think this is good.

I'll open another issue to support `STDOUT` since passing a string won't work.